### PR TITLE
Enhancements to the ISmsClient

### PR DESCRIPTION
**Pull request by guctum**
Addresses #222

Added overload method to ISmsClient that skips initializing a separate request object

Please let me know if there's anything missing, I'm very new to open source so am interested in learning more as well as contributing quality code.

### CODE REVIEW

File: Vonage/Messaging/SmsClient.cs [hunk 1]
```diff
@@ -41,5 +41,15 @@ private static void ValidSmsResponse(SendSmsResponse smsResponse)
                 throw new VonageSmsResponseException($"SMS Request Failed with status: {smsResponse.Messages[0].Status} and error message: {smsResponse.Messages[0].ErrorText}") { Response = smsResponse};
             }
         }
+
+        SendSmsResponse ISmsClient.SendAsSms(string to, string from, string text, SmsType smsType, Credentials credentials)
+        {
+            var result = ApiRequest.DoPostRequestUrlContentFromObject<SendSmsResponse>(
+                ApiRequest.GetBaseUri(ApiRequest.UriType.Rest, "sms/json"),
```
Comment (slorello89) [inline Vonage/Messaging/SmsClient.cs:48]: This needs to pass a SendSmsRequest object into the ApiRequest class.

**Review by slorello89 — CHANGES_REQUESTED** [review]
Hi @guctum - first let me thank you for opening a PR - welcome to the world of open source :). A couple of things I'd ask from you before we can move forward.

1: The way you're calling the `ApiRequest` library won't work in this case, the `ApiRequest` class's `DoPostRequestUrlContentFromObject` is expecting an object to be passed in - your code compiles because `Credentials` is in fact an object, however, it's not the request object. If you look at the `SendAnSms` method - you'll see what I'm talking about, the `request` is being passed into the `ApiRequest` method. That's what the `ApiRequest` method will convert into a request body. To fix this, you'll just need to initialize an instance of `SendSmsRequest`, and pass it into the `ApiRequest`.

2. You'll notice CodeCov complaining about a decrease in code coverage. This library has pretty good test coverage (~90%) - and most of what's not covered has been deprecated. Before this is merged we just want to make sure that we have tests establishing that it works as intended. You can see the current tests available in the [MessagingTests](https://github.com/Vonage/vonage-dotnet-sdk/blob/master/Vonage.Test.Unit/MessagingTests.cs) class for inspiration, but essentially you are going to have a request body that you are expecting to see at a given URI, and it will respond with a response structure that you define.

Thanks again for contributing, looking forward to getting this merged!

**Comment by guctum** [comment]
Hey @slorello89 didn't want you to think I forgot about this! My day job got a bit crazy, so I haven't had time to go through and knock out the unit tests. Thanks for the pointer on my code, I did work with that I just need to do the tests. As I've never done unit testing in C# before, I need to sit down (likely this weekend) and familiarize myself with the testing library.

**Comment by slorello89** [comment]
No worries @guctum totally understandable and I commend you for contributing :). Not much you need to work out if you have the [.NET Core SDK](https://dotnet.microsoft.com/download) installed. Just need to run `dotnet test` to run everything. The tests you'll need to write are going to pretty much be built off the boilerplate of the ones in the [MessagingTests](https://github.com/Vonage/vonage-dotnet-sdk/blob/master/Vonage.Test.Unit/MessagingTests.cs) class that I pointed you at earlier.

**Comment by slorello89** [comment]
Going to close this PR as the contest has ended and I've not seen an update on it for a while - thanks again for opening the PR!

**Pull request closed without merging**